src/agent/LibFuzzerDotnetLoader/Program.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using System.Runtime;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using TestOneSpan = SharpFuzz.ReadOnlySpanAction;
+delegate void TestOneArray(byte[] data);
+
+namespace LibFuzzerDotnetLoader {
+    public class Program {
+        public static void Main(string[] args) {
+            var target = Environment.GetEnvironmentVariable("LIBFUZZER_DOTNET_TARGET");
+
+            if (target == null) {
+                throw new Exception("`LIBFUZZER_DOTNET_TARGET` not set. " +
+                                    "Expected format: \"<assembly-path>:<class>:<static-method>\"");
+            }
+
+            var parts = target.Split(':');
+
+            var assemPath = parts[0];
+            var typeName = parts[1];
+            var methodName = parts[2];
+
+            var assem = Assembly.LoadFrom(assemPath);
+
+            var ty = assem.GetType(typeName);
+
+            if (ty == null) {
+                throw new Exception($"unable to resolve type: {typeName}");
+            }
+
+            var method = ty.GetMethod(methodName);
+
+            if (method == null) {
+                throw new Exception($"unable to resolve method: {methodName}");
+            }
+
+            if (TryTestOneSpan(method)) {
+                return;
+            }
+
+            if (TryTestOneArray(method)) {
```

Comment:
We support multiple test function signatures, to support projects that target pre-`Span` SDKs.

Reply:
I suggest printing out a warning message when we detect an old version. to let the user know that this might not be the most optimal route ie. there will be an extra allocation for each call of their test function

Reply:
Will add that in a follow-up PR.

Reply:
Tracked in #2129.